hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py

```diff
@@ -1084,20 +1061,22 @@ async def request(self,
                     query += f"&signature={signature}"
```

Comment:
Authentication should be in a separate object (approach taken by the newer connectors — e.g. ByBit). Also, the auth responsibility should be pushed in the `WebAssistants` by inheriting from `AuthBase`. An example of that is CoinbasePro

Reply:
AuthBase defines rest_authenticate and ws_authenticate. rest_authenticate in the CoinbasePro's implementation extends headers. In case of Binance Perpetual there's no real authantication going on, and headers are not modified, a signature is just added to the URL.